cmd/vela-s3-cache/flush.go

```diff
@@ -19,63 +19,92 @@ const flushAction = "flush"
+	objectsExist := false
 
-	// Indicate to our routine to exit cleanly upon return.
-	defer close(doneCh)
+	// set a timeout on the request to the cache provider
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	logrus.Debugf("listing objects in bucket %s under path %s", f.Root, f.Path)
+	logrus.Infof("processing cached objects in path %s", f.Namespace)
 
-	// lists all objects matching the path ib
-	// the specified bucket
-	objectCh := mc.ListObjectsV2(f.Root, f.Path, true, doneCh)
+	opts := minio.ListObjectsOptions{
+		Prefix:    f.Namespace,
+		Recursive: true,
+	}
+	// lists all objects matching the path
+	// in the specified bucket
+	objectCh := mc.ListObjects(ctx, f.Root, opts)
 	for object := range objectCh {
+		// we got at least one object
+		objectsExist = true
+
 		if object.Err != nil {
 			return fmt.Errorf("unable to retrieve object %s: %s", object.Key, object.Err)
 		}
 
-		// Check if the object meets the flush age
-		if object.LastModified.Before(time.Now().AddDate(0, 0, f.Age*-1)) {
-			logrus.Debugf("removing object from bucket %s in path: %s", f.Root, f.Path)
+		logrus.Infof("  - %s; last modified: %s", object.Key, object.LastModified.String())
+
+		// determine time in the past for flush cut off
+		timeInPast := time.Now().Add(-f.Age)
+
+		// check if the object meets the flush age
+		if object.LastModified.Before(timeInPast) {
+			logrus.Infof("    ├ '%s' flush age criteria met. removing object.", f.Age)
 
 			// remove the object from the bucket
-			err := mc.RemoveObject(f.Root, fmt.Sprintf("%s/%s", f.Path, object.Key))
+			err := mc.RemoveObject(ctx, f.Root, object.Key, minio.RemoveObjectOptions{})
 			if err != nil {
 				return err
 			}
+
+			// verify that the object is gone, .RemoveObject fails silently
+			// if the supplied path leads to an object that doesn't exist
+			_, err = mc.StatObject(ctx, f.Root, object.Key, minio.StatObjectOptions{})
+			if err != nil {
+				logrus.Info("    ├ object successfully removed.")
+			} else {
+				return fmt.Errorf("object %s was not removed: %v", object.Key, err)
+			}
+		} else {
+			logrus.Infof("    ├ '%s' flush age criteria not met. keeping object.", f.Age)
 		}
 	}
 
+	if !objectsExist {
+		logrus.Infof("no cache objects found at %s", f.Path)
+	}
+
+	logrus.Infof("cache flush action completed")
+
 	return nil
 }
 
-// Configure prepares the flush fields for the action to be taken
-func (f *Flush) Configure(repo Repo) error {
+// Configure prepares the flush fields for the action to be taken.
+func (f *Flush) Configure(repo *Repo) error {
 	logrus.Trace("configuring flush action")
 
-	// set the default prefix of where to save the object
-	path := fmt.Sprintf("%s/%s/%s", strings.TrimRight(f.Path, "/"), repo.Owner, repo.Name)
-
-	if len(f.Path) > 0 {
-		path = fmt.Sprintf("%s/%s/%s/%s", strings.TrimRight(f.Prefix, "/"), repo.Owner, repo.Name, f.Path)
-	}
+	// construct the object path
+	path := buildNamespace(repo, f.Prefix, f.Path, "")
```

Comment:
created a utility function to create the object path to share between the different actions for consistency